compiler/quilt/tools/package.py

```diff
@@ -74,6 +74,33 @@ def __init__(self, store, user, package, path, contents=None, pkghash=None):
 
         self._contents = contents
 
+    def __getitem__(self, item):
+        node = self.get_contents()
+
+        if '/' in item and '.' in item:
+            raise ValueError("Invalid name: expected delimiter of '.' or '/', not both for {!r}".format(item))
+        if '.' in item:
+            parts = item.split('.')
+        elif '/' in item:
+            parts = item.split('/')
+        else:
+            parts = [item]
+
+        try:
+            for part in parts[:-1]:
```

Comment:
This seems overly complicated. Why not: for part in parts...return node?

Reply:
You're right.